src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,85 @@
+package duke;
+
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+/**
+ * Parser class extracts tokens from a String like a Scanner, intended for used for parsing arguments given to a
+ * command.
+ */
+public class Parser {
+    private final String[] strs;
+    private int cursor = 0;
+    
+    /**
+     * Constructs a Parser object with the provided input string. read operations on this parser reads tokens from the
+     * String as if it were an array of space-separated strings.
+     *
+     * @param str The input string.
+     */
+    public Parser(String str) {
+        this.strs = str.split(" ");
+    }
+    
+    /**
+     * consumes a token and returns it.
+     *
+     * @return The next string.
+     * @throws DukeOptionParsingException If all tokens have been read from the input
+     */
+    public String next() throws DukeOptionParsingException {
+        String ret = this.peek();
```

Comment:
Perhaps a more intuitive variable name here?

Reply:
I think the name is fine for now given the size of the code block, though I'll rename it to `token` for readability, in case the complexity of the code increases. Thanks!